parsing/mod_adapt.py

```diff
@@ -0,0 +1,104 @@
+"""
+This module contains functionality for extracting a grammar from
+classes in a module.
+"""
+import types
+from parsing.grammar import Precedence, TokenSpec, NontermSpec, SpecError
+from parsing.ast_objs import Token, Nonterm
+
+class ModuleAdapter(object):
+    def __init__(self, modules):
+        if type(modules) == types.ModuleType:
+            # Wrap single module in a list.
+            modules = [modules]
+        self.modules = modules
+        items = []
+        for module in self.modules:
+            for k, v in module.__dict__.items():
+                if isinstance(v, type) and isinstance(v.__doc__, str):
+                    dirtoks = v.__doc__.split(" ")
+                    items.append((module, k, v, dirtoks))
+        self.named_objs = items
+
+    def get_precedences(self):
+        result = []
```

Comment:
This computation is relatively expensive, so let's either precompute this in `__init__`, or cache the results so that subsequent calls don't recompute.  Ditto for `get_tokens` and `get_nonterminals`.